### ISSUE TITLE
go-getter parameters bug for helm chart url

### ISSUE THREAD

**Issue by jackivanov** [rancher/fleet#428]
Fleet adds extra question mark to the helm repo url

```
namespace: test
helm:
  chart: git::ssh://git@test.com/test/fleet-test.git?ref=fleet
```

This gitrepo fails with error:

```
time="2021-06-21T15:39:28Z" level=fatal msg="error downloading 'ssh://git@test.com/test/fleet-test.git?ref=fleet?sshkey=xxx
```

**?sshkey** should be **&sshkey**

related code https://github.com/rancher/fleet/blob/3b1da5727d9f4ee3e562cbfceb3032fcd44c9808/pkg/bundle/resources.go#L340-L344

**Comment by ron1** [comment]
@StrongMonkey @nickgerace Would be great to get this simple fix in for the 0.3.6 release.

**Comment by nickgerace** [comment]
Marked as a Release Candidate in the hopes of getting it scheduled sooner rather than later. Thank you for the issue.

**Comment by StrongMonkey** [comment]
Available to test on fleet v0.3.8-rc3.

**Comment by prachidamle** [comment]
Validated that even if the helm chart url has query paramenters then the gitrepo does not error out in clone/deployment.

Test Configuration
-----------------------
Rancher v2.6-head
Fleet v0.3.8
Single node docker standalone cluster

Validation Method
---------------------
1) I duplicated the fleet-examples git repository and created a new private repository for testing
2) Configured the repository to use SSH auth. I generated a new SSH key pair in the following  way:

`ssh-keygen -t rsa -b 4096 -m pem -C "izaac@email.com" (No passphrase)`

3) Edited the fleet.yaml in this example to contain a query parameter in the chart URL:
https://github.com/prachidamle/fleet-examples/blob/master/multi-cluster/helm-external/fleet.yaml

```
namespace: fleet-mc-helm-external-example
helm:
  chart: https://github.com/rancher/fleet-examples/releases/download/example/guestbook-0.0.0.tgz?prachi=true
```

4) Created a GitRepo in the Cluster Explorer UI using the ssh auth
5) GitRepo was synced successfully and saw bundles deployed.
